packages/ariakit-react-core/src/utils/store.tsx

```diff
@@ -63,9 +63,14 @@ function safeFlushSync(fn: AnyFunction, canFlushSync = true) {
   inFlushSyncContext = true;
   const originalError = console.error;
   if (process.env.NODE_ENV !== "production") {
-    console.error = (msg: string) => {
-      if (msg.startsWith("Warning: flushSync")) return;
-      originalError(msg);
+    console.error = (...data: any[]) => {
+      if (
+        typeof data[0] === "string" &&
+        data[0].startsWith("Warning: flushSync")
+      ) {
+        return;
+      }
+      originalError(data);
```

Comment:
Good catch! Thanks for the fix.

Shouldn't the ﻿`data` be spread into the original error function?

```suggestion
      originalError(...data);
```

Reply:
Yes sir, my bad on that.